src/main/webapp/definitions/alfresco-model.yaml

```diff
@@ -161,7 +159,143 @@ paths:
           description: Unexpected error
           schema:
             $ref: '#/definitions/Error'
+  '/types':
+    get:
+      x-alfresco-since: "7.0.0"
+      tags:
+        - types
+      summary: List types
+      description: |
+        **Note:** This is available in Alfresco 7.0.0 and newer versions.
+
+        Gets a list of types from the data dictionary. The System types will be ignored by default.
+        ```JSON
+        {
+          "list": {
+            "pagination": {
+              "count": 0,
+              "hasMoreItems": true,
+              "totalItems": 0,
+              "skipCount": 0,
+              "maxItems": 0
+            },
+            "entries": [
+              {
+                "entry": {
+                  "id": "cm:content",
+                  "description": "Base Content Object",
+                  "title": "Content",
+                  "parentId": "cm:cmobject"
+                  "properties": [
+                    {
+                      "id": "cm:name",
+                      "title": "Name",
+                      "description": "Name",
+                      "dataType": "d:text",
+                      "isMultiValued": false,
+                      "isMandatory": true,
+                      "isMandatoryEnforced": true
+                      "isProtected": false
+                      ...
+                    },
+                    {
+                      ...
+                    }
+                  ]
+                }
+              },
+              {
+                "entry": {
+                  ...
+                }
+              },
+              {
+                "entry": {
+                  ...
+                }
+              },
+            ]
+          }
+        }
+        ```
+      operationId: listTypes
+      parameters:
+        - name: where
+          in: query
+          description: |
+            Optionally filter the list. Here are some examples:
+            A Type should represented in the fully qualified name(`namespace_prefix:name`). e.g 'cm:content'.
 
+            The following where clause will only return types from the `namespace1:model` and `namespace2:model`.
+              ```
+              where=(modelIds='namespace1:model,namespace2:model')
+              ```
+
+            The following where clause will only return sub types for the given parents.
+              ```
+              where=(parentIds='namespace1:parent,namespace2:parent')
+              ```
+
+            The following where clause will only return types that match the pattern.
+              ```
+              where=(uriPrefix matches('http://www.alfresco.org/model.*'))
```

Comment:
Jan had a good point about this in the demo last week. 

"uriPrefix" isn't really representative of what you're checking, I believe it's actually "namespaceUri" as defined here: https://github.com/Alfresco/alfresco-community-repo/blob/ba9ae7071aab2225391ff8c83c91629a9118529c/data-model/src/main/java/org/alfresco/service/namespace/QName.java#L50

Furthermore, whatever properties we refer to in the where clause should be returned in the response otherwise how will users know what values they can use to filter by? Currently we don't return any model information at all.

I will add some comments to the RFC document to discuss further.